docs/throttle.md

```diff
@@ -0,0 +1,101 @@
+2. Total voting power of a chain changes over time, especially as validators are jailed/tombstoned. As validators are jailed, total voting power decreases, and so does the slashing allowance per replenishment period.
+3. The voting power allowance added to the slash meter per replenish period will always be strictly greater than 1. If the `SlashMeterReplenishFraction (param)` is set too low, integer rounding will put this minimum value into effect.
+
+The behavior described above is achieved by executing `CheckForSlashMeterReplenishment()` every endblock.
+
+## Throttling Invariant
+
+Using on-chain params and the sub protocol defined, slash packet throttling is implemented such that the following invariant is maintained (in addition to those already defined in the CCV spec).
+
+For the following invariant to hold, these points must be true:
+
+- The final slashed validator does not have more than `SlashMeterReplenishFraction` of total voting power on the provider.
+- `SlashMeterReplenishFraction` is large enough to avoid rounding errors.
+- `SlashMeterReplenishPeriod` is sufficiently longer than the time it takes to produce a block.
+
+Invariant:
+
+> The time it takes to jail `X`% of the provider validator set from consumer initiated slash requests will be greater than or equal to `(X * SlashMeterReplenishPeriod / SlashMeterReplenishFraction) - 2 * SlashMeterReplenishPeriod`
+
+Intuition: If jailings begin when the slash meter is full, then `SlashMeterReplenishFraction` of the provider validator set can be jailed immediately. The remaining jailings are only applied when the slash meter is positive in value, so the time it takes to jail the remaining `X - SlashMeterReplenishFraction` of the provider validator set is `(X - SlashMeterReplenishFraction) * SlashMeterReplenishPeriod / SlashMeterReplenishFraction`. However, the final validator could be jailed during the final replenishment period, with the meter being very small in value (causing it to go negative after jailing). So we subtract another `SlashMeterReplenishPeriod` term in the invariant to account for this.
+
+Note this invariant could be adjusted with different slash meter protocols, but the current scheme is the simplest to implement and understand.
+
+This invariant is useful because it allows us to reason about the time it takes to jail a certain percentage of the provider validator set from consumer initiated slash requests. For example, if `SlashMeterReplenishFraction` is set to 0.06, then it takes no less than 4 replenishment periods to jail 33% of the provider validator set on the Cosmos Hub. Note that as of writing this on 11/29/22, the Cosmos Hub does not have a validator with more than 6% of total voting power.
```

Comment:
> no less than 4 replenishment periods to jail 33% of the provider validator set 

The same issue applies as above, the validator set is changing over time.

Reply:
doc has been updated to refer to the initial validator set